plugins/filter/merge.py

```diff
@@ -53,6 +53,9 @@ def merge_with_app(base, new):
             # Use the list(set()) trick to remove duplicated items
             base_service[k] = sorted(list(set(new_service[k] + base_service[k])))
 
+        if new_service.get("environment_variables") is not None:
+            base_service["environment_variables"] = new_service["environment_variables"]
```

Comment:
You are in a `for` loop so shouldn't you aggregate all services like it is done for other items above?